### ISSUE TITLE
Uncaught TypeError: Cannot read property 'position' of undefined

### ISSUE THREAD

**Issue by mrstebo** [AtomLinter/atom-minimap-linter#25]
[Enter steps to reproduce:]

1. Restart Atom with a file open (so that it appears again after the restart). This error doesn't happen all the time 😕 

**Atom**: 1.20.1 x64
**Electron**: 1.6.9
**OS**: Mac OS X 10.12.6
**Thrown From**: [minimap-linter](https://github.com/AtomLinter/atom-minimap-linter) package 2.1.0

### Stack Trace

Uncaught TypeError: Cannot read property 'position' of undefined

```
At /Users/steve/.atom/packages/minimap-linter/lib/minimap-linter.js:10

TypeError: Cannot read property 'position' of undefined
    at messageRange (/packages/minimap-linter/lib/minimap-linter.js:10:52)
    at goodMessage (/packages/minimap-linter/lib/minimap-linter.js:12:41)
    at /packages/minimap-linter/lib/minimap-linter.js:93:17
    at Array.forEach (native)
    at /packages/minimap-linter/lib/minimap-linter.js:92:30
    at Map.forEach (native)
    at Object.render (/packages/minimap-linter/lib/minimap-linter.js:80:25)
    at /packages/linter/lib/ui-registry.js:32:16
    at Set.forEach (native)
    at UIRegistry.render (/packages/linter/lib/ui-registry.js:31:20)
    at /packages/linter/lib/main.js:187:23
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at MessageRegistry.update (/packages/linter/lib/message-registry.js:130:20)
    at later (/packages/linter/node_modules/sb-debounce/index.js:9:14)
```

### Commands

```
     -0:14.7.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-when-scrolling)
     -0:13.7.0 intentions:highlight (input.hidden-input)
```

### Non-Core Packages

```
busy-signal 1.4.3 
docker 0.8.0 
file-icons 2.1.12 
highlight-selected 0.13.1 
intentions 1.1.5 
language-docker 1.1.8 
linter 2.2.0 
linter-docker 0.2.1 
linter-handlebars 2.5.0 
linter-rubocop 2.1.1 
linter-swagger 0.4.2 
linter-ui-default 1.6.10 
minimap 4.29.7 
minimap-autohide 0.10.1 
minimap-bookmarks 0.4.2 
minimap-cursorline 0.2.0 
minimap-find-and-replace 4.5.2 
minimap-git-diff 4.3.1 
minimap-highlight-selected 4.6.1 
minimap-linter 2.1.0 
minimap-pigments 0.2.2 
minimap-selection 4.5.0 
pretty-json 1.6.4 
toggle-quotes 1.0.1 
```

**Comment by mrstebo** [comment]
Also happens when clicking to another part of a file. It is only happening on YAML files, which maybe something to do with `linter-swagger` 🤔

**Comment by Arcanemagus** [comment]
Looks like `linter-swagger` is passing an invalid message up that somehow is getting past the checks in `linter` itself.

**Comment by Arcanemagus** [comment]
Fix published in v2.1.1, thanks for the report! (Now to fix `linter-swagger` so it doesn't send invalid messages in the first place...)

**Comment by mrstebo** [comment]
Great work! I was going to have a crack at this myself, but couldn't find any time 😅
